src/rom3.c

```diff
@@ -0,0 +1,435 @@
+#include "global.h"
+#include "asm.h"
+#include "items.h"
+#include "link.h"
+#include "pokemon.h"
+#include "species.h"
+#include "task.h"
+
+extern u16 gBattleTypeFlags;
+
+extern u32 gUnknown_020239FC;
+extern u32 gUnknown_02024A64;
+extern u8 gUnknown_02024A68;
+extern u16 gUnknown_02024A6A[];
+extern u8 gUnknown_02024A72[];
+extern u8 gUnknown_02024C78;
+extern u8 gUnknown_02024E60[];
+extern u8 gUnknown_02024E64[];
+extern void (*gUnknown_030042D4)(void);
+extern void (*gUnknown_03004330[])(void);
+
+extern void sub_800BF28(void);
+extern void sub_8083C50(u8);
+extern void nullsub_41(void);
+extern void nullsub_91(void);
+extern void battle_anim_clear_some_data(void);
+extern void sub_8040710(void);
+extern void BattleAI_HandleItemUseBeforeAISetup(void);
+extern void sub_8094978(u8, int);
+extern void sub_800BA78(void);
+extern void sub_800B9A8(void);
+extern void sub_800BD54(void);
+extern void sub_8010800(void);
+extern void sub_812B468(void);
+extern void sub_8137224(void);
+extern void sub_802BF74(void);
+extern void sub_8032AE0(void);
+extern void sub_8037510(void);
+extern void sub_811DA78(void);
+
+void sub_800B858(void)
+{
+    if (gBattleTypeFlags & 2)
+    {
+        OpenLink();
+        CreateTask(sub_8083C50, 0);
+        sub_800BF28();
+    }
+}
+
+void setup_poochyena_battle(void)
+{
+    s32 i;
+    
+    gUnknown_030042D4 = nullsub_41;
+    for (i = 0; i < 4; i++)
+    {
+        gUnknown_03004330[i] = nullsub_91;
+        gUnknown_02024A72[i] = 0xFF;
+        gUnknown_02024E60[i] = 0;
+        gUnknown_02024E64[i] = 0;
+    }
+    sub_800B858();
+    gUnknown_02024A64 = 0;
+    battle_anim_clear_some_data();
+    sub_8040710();
+    BattleAI_HandleItemUseBeforeAISetup();
+    if (gBattleTypeFlags & 0x10)
```

Comment:
`if (gBattleTypeFlags & BATTLE_TYPE_FIRST_BATTLE)`

This function checks if the current battle is the Poochyena battle, but it may be used in other contexts as well.